File: python/reverse_integer.py
#!/usr/bin/env python

class Solution:
    def reverse(self, x):
        sign = 1
        if not x: # negative number
            sign, x = -1, abs(x)

        result = 0
        while x:
            result = result*10 + x%10
            x /= 10

        # for a 32 bits integer, the max value is (2^31 - 1)
        if result >= (2**31 - 1):
            return 0

        return sign*result

File: java/ReverseInteger.java
/**
 * Created by Robbert on 1/25/15.
 * Reverse Integer
 * https://oj.leetcode.com/problems/reverse-integer/
 */
public class ReverseInteger {
    public int reverse(int x) {
        int result = 0;

        while (x != 0) {
            if (Math.abs(result) > Integer.MAX_VALUE / 10) {
                return 0;
            } else {
                result = result * 10 + x % 10;
            }
            x /= 10;
        }

        return result;
    }
}
